spline-shadow/client-ui/src/app/store/reducers/execution-plan.reducer.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as ExecutionPlanAction from '../actions/execution-plan.actions';
+import { ExecutedLogicalPlanVM } from '../../model/viewModels/executedLogicalPlanVM';
+import { OperationType } from '../../model/types/operationType';
+
+export type Action = ExecutionPlanAction.ExecutionPlanActions
+
+export function executionPlanReducer(state: ExecutedLogicalPlanVM, action: Action) {
+    switch (action.type) {
+        case ExecutionPlanAction.ExecutionPlanActionTypes.EXECUTION_PLAN_GET_SUCCESS: return { ...state, ...action.payload }
+        default: return state
+    }
+}
+
+
+export function getIconFromOperationType(operation: string): number {
+    switch (operation) {
+        case OperationType.Projection: return 0xf13a
+        case OperationType.BatchRead: return 0xf085
+        case OperationType.LogicalRelation: return 0xf1c0
+        case OperationType.StreamRead: return 0xf085
+        case OperationType.Join: return 0xf126
+        case OperationType.Union: return 0xf0c9
+        case OperationType.Generic: return 0xf0c8
+        case OperationType.Filter: return 0xf0b0
+        case OperationType.Sort: return 0xf161
+        case OperationType.Aggregate: return 0xf1ec
+        case OperationType.WriteCommand: return 0xf0c7
+        case OperationType.BatchWrite: return 0xf0c7
+        case OperationType.StreamWrite: return 0xf0c7
+        case OperationType.Alias: return 0xf111
+        default: return 0xf15b
```

Comment:
Don't you want to create a static map/object like this:
```typescript
const operationIconCodes: Map<string, number> = new Map([
  [OperationType.Projection, 0xf13a],
  [OperationType.BatchRead, 0xf085],
  ...
  [OperationType.Generic, 0xf15b]  // <-- default icon
])
...
operationIconCodes.get(operationType)
  || operationIconCodes.get(OperationType.Generic)
```

Reply:
Yes, that's better